subnet/deployment-generator/script/gen_compose.js

```diff
@@ -0,0 +1,162 @@
+
+const net = require('net');
+const config = require('./config_gen')
+Object.freeze(config)
+
+module.exports = { 
+  genSubnetNodes,
+  genBootNode,
+  genObserver,
+  genServices,
+  genComposeEnv,
+  injectMacConfig
+};
+
+
+function genSubnetNodes(machine_id, num, start_num=1) {
+  subnet_nodes = {}
+  for (let i=start_num; i < start_num+num; i++) {
+    node_name='subnet'+i.toString()
+    volume='./xdcchain'+i.toString()+':/work/xdcchain'
+    var config_path='${SUBNET_CONFIG_PATH}/subnet'+i.toString()+'.env'
+    compose_profile='machine'+machine_id.toString()
+    port = 20303+i-1
+    rpcport = 8545+i-1
+    wsport= 9555+i-1
```

Comment:
why -1?

Reply:
because 'i' starts at 1 in this loop.

Reply:
then can you start with 20302?

Reply:
sure, just thought it's easier to read when number is the same as start port